consensus/src/user/provisioners.rs

```diff
@@ -12,17 +12,55 @@ use node_data::bls::{PublicKey, PublicKeyBytes};
 use node_data::ledger::Seed;
 use num_bigint::BigInt;
 use std::collections::BTreeMap;
+use std::mem;
 
 use super::committee::Committee;
 
 pub const DUSK: u64 = 1_000_000_000;
 
-#[derive(Clone, Default, Debug)]
+#[derive(Clone, Debug)]
 pub struct Provisioners {
     members: BTreeMap<PublicKey, Stake>,
 }
 
+#[derive(Clone, Debug)]
+pub struct ContextProvisioners {
+    current: Provisioners,
+    prev: Option<Provisioners>,
```

Comment:
Not sure how feasible would be, but do you think we can re-implement this in future by having a delta? That would be helpful if we maintain a large number of provisioners. For now, it's just an idea.

```
pub struct ContextProvisioners {
    current: Provisioners,
    delta: Option<Provisioners, (DELETED|ADDED)>, 
```

This will be also much cheaper on persisting it.

Reply:
Current idea is to not persist those anymore.

For the other concern, we should bench it before